scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -361,7 +363,7 @@ public final DTLSFlight processMessage(Record record) throws HandshakeException
 			try {
 				record.setSession(session);
 				DTLSMessage messageToProcess = inboundMessageBuffer.getNextMessage(record);
-				while (nextFlight == null && messageToProcess != null) {
+				while (messageToProcess != null) {
```

Comment:
I see a possible different behavior.
In the case where we find a complete flight and there is still some messageToProcess, we will continue to loop even if we already send the flight. That was not the case before.
I suppose this is not really an issue ?